### PR TITLE
Fixed statistics for online contest

### PR DESCRIPTION
**Pull request by gogo4ds**
Maximum Results in online contests statistics not calculated correctly
Closes https://github.com/SoftUni-Internal/suls-issues/issues/4708

### CODE REVIEW

File: Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs [hunk 1]
```diff
@@ -360,33 +360,69 @@ public ActionResult Stats(int contestId, bool official)
...
             var toPoints = 0;
-            foreach (var problem in contestResults.Problems)
+
+            if (contest.IsOnline && official)
+            {
+                statsModel.IsGroupedByProblemGroup = true;
+
+                foreach (var problems in contestResults.Problems.GroupBy(p => p.ProblemGroupId))
+                {
+                    var maxResultsForProblemGroupCount = contestResults.Results
```
Comment (achobanov) [inline Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs]: Consider extracting in data service.

File: Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs [hunk 2]
```diff
@@ -360,33 +360,69 @@ public ActionResult Stats(int contestId, bool official)
...
+
+                    AddStatsByProblem(problemGroupName, maxPointsForProblemGroup, maxResultsForProblemGroupCount);
+                    AddStatsByPointsRange(maxPointsForProblemGroup, contestResults.Results);
+                }
+            }
+            else
+            {
+                foreach (var problem in contestResults.Problems)
+                {
+                    var maxResultForProblemCount = contestResults.Results
```
Comment (achobanov) [inline Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs]: Consider extracting in data service.

File: Open Judge System/Web/OJS.Web/Areas/Users/Views/Shared/_ProfileInfo.cshtml [hunk 3]
```diff
@@ -146,9 +146,9 @@
                     @if (userIsProfileOwnerOrAdmin)
                     {
                         <td>
-                            @Resource.Compete: <span class="text-white">@result.CompeteResult / @result.ContestMaximumPoints.GetValueOrDefault()</span>
+                            @Resource.Compete: <span class="text-white">@result.CompeteResult / @result.ContestCompeteMaximumPoints.GetValueOrDefault()</span>
```
Comment (achobanov) [inline Open Judge System/Web/OJS.Web/Areas/Users/Views/Shared/_ProfileInfo.cshtml]: Consider extracting into variable.

File: Open Judge System/Web/OJS.Web/Areas/Users/Views/Shared/_ProfileInfo.cshtml [hunk 4]
```diff
@@ -146,9 +146,9 @@
...
                             <br />
-                            @Resource.Practice: <span class="text-white">@result.PracticeResult / @result.ContestMaximumPoints.GetValueOrDefault()</span>
+                            @Resource.Practice: <span class="text-white">@result.PracticeResult / @result.ContestPracticeMaximumPoints.GetValueOrDefault()</span>
```
Comment (achobanov) [inline Open Judge System/Web/OJS.Web/Areas/Users/Views/Shared/_ProfileInfo.cshtml]: Consider extracting into variable.

File: Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs (hunk 1 above)
Comment (gogo4ds) [reply Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs]: I operate on already queried set of results from a view model, it is not for data service

File: Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs (hunk 2 above)
Comment (gogo4ds) [reply Open Judge System/Web/OJS.Web/Areas/Contests/Controllers/ResultsController.cs]: I operate on already queried set of results from a view model, it is not for data service

**Pull request merged**